pymoto/modules/assembly.py

```diff
@@ -245,28 +281,72 @@ class AssembleMass(AssembleGeneral):
 
     def _prepare(self, domain: DomainDefinition, *args, rho: float = 1.0, bcdiagval=0.0, **kwargs):
         # Element mass matrix
-        # 1/36 Mass of one element
-        mel = rho * np.prod(domain.element_size)
+        self.ME = ConsistentMassEq(domain, ndof=domain.dim, MP=rho)
+        super()._prepare(domain, self.ME, *args, bcdiagval=bcdiagval, **kwargs)
 
-        if domain.dim == 2:
-            # Consistent mass matrix
-            ME = mel / 36 * np.array([[4.0, 0.0, 2.0, 0.0, 2.0, 0.0, 1.0, 0.0],
-                                      [0.0, 4.0, 0.0, 2.0, 0.0, 2.0, 0.0, 1.0],
-                                      [2.0, 0.0, 4.0, 0.0, 1.0, 0.0, 2.0, 0.0],
-                                      [0.0, 2.0, 0.0, 4.0, 0.0, 1.0, 0.0, 2.0],
-                                      [2.0, 0.0, 1.0, 0.0, 4.0, 0.0, 2.0, 0.0],
-                                      [0.0, 2.0, 0.0, 1.0, 0.0, 4.0, 0.0, 2.0],
-                                      [1.0, 0.0, 2.0, 0.0, 2.0, 0.0, 4.0, 0.0],
-                                      [0.0, 1.0, 0.0, 2.0, 0.0, 2.0, 0.0, 4.0]])
-        elif domain.dim == 3:
-            ME = np.zeros((domain.elemnodes*domain.dim, domain.elemnodes*domain.dim))
-            weights = np.array([8.0, 4.0, 2.0, 1.0])
-            for n1 in range(domain.elemnodes):
-                for n2 in range(domain.elemnodes):
-                    dist = round(np.sum(abs(np.array(domain.node_numbering[n1]) - np.array(domain.node_numbering[n2])))/2)
-                    ME[n1*domain.dim+np.arange(domain.dim), n2*domain.dim+np.arange(domain.dim)] = weights[dist]
-
-            ME *= mel / 216
-        else:
-            raise RuntimeError("Only for 2D and 3D")
-        super()._prepare(domain, ME, *args, bcdiagval=bcdiagval, **kwargs)
+
+class AssembleScalarMass(AssembleGeneral):
+    r""" Consistent mass equivalent matrix assembly by scaling elements (e.g. Thermal Capacity)
+        :math:`\mathbf{C} = \sum_e x_e \mathbf{C}_e`
+
+        Input Signal:
+            - ``x``: Scaling vector of size ``(Nel)``
+
+        Output Signal:
+            - ``C``: Scalar mass equivalent matrix of size ``(n, n)``
+
+        Args:
+            domain: The domain to assemble for -- this determines the element size and dimensionality
+            *args: Other arguments are passed to AssembleGeneral
+
+        Keyword Args:
+            MP: Material property for element matrix (e.g. Thermal capacity)
+            bcdiagval: The value to put on the diagonal in case of boundary conditions (bc)
+            **kwargs : Other keyword-arguments are passed to AssembleGeneral
+        """
+
+    def _prepare(self, domain: DomainDefinition, *args, MP: float = 1.0, bcdiagval=0.0, **kwargs):
+        # Element mass equivalent matrix
+        self.CE = ConsistentMassEq(domain, ndof=1, MP=MP)
+        super()._prepare(domain, self.CE, *args, bcdiagval=bcdiagval, **kwargs)
+
+
+class AssembleScalarField(AssembleGeneral):
```

Comment:
For the naming of this class we probably need to see how the PDE looks for this discretization. A scalar-field might throw users off, since mass is also a scalar field, but needs another class. @artofscience do you have some ideas for another name (something like ∇.∇u but then with a suitable name?)

Reply:
Cook calls it a Scalar Field Element, issue is that it is applicable to a multitude of analyses, so a general name will always be slightly confusing/obscure

Reply:
I am not sure. Indeed, to generalize, we probably want to follow some categorization of pde's and name according to that.

Reply:
Something referencing Poisson's equation then?

Reply:
It it is a Poisson equation, then you could call it so. Might be nice to also add the Cook reference (page number etc) to the docstring

Reply:
If I remember correctly it is Poisson's equation for steady state heat transfer and electrostatics

Reply:
called it AssemblePoisson for now